File: codes/mpi_hello.f90
program main

  use mpi
  implicit none
  integer :: ierr, rank, size

  call MPI_INIT(ierr)
  call MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierr)
  call MPI_COMM_SIZE(MPI_COMM_WORLD, size, ierr)
  print *, "Hello, world! I am process ", rank, " of ", size
  call MPI_FINALIZE(ierr)

end program main

File: codes/mpi_hello.cpp
/*
  The entire set of C++ language bindings have been deprecated since MPI-2.2
*/
#include "mpi.h"
#include <iostream>
using namespace std;

int main(int argc, char* argv[])
{
  int rank, size;

  MPI::Init(argc, argv);
  rank = MPI::COMM_WORLD.Get_rank();
  size = MPI::COMM_WORLD.Get_size();  
  cout << "Hello, world! I am process " << rank << " of " << size << endl;
  MPI::Finalize();

  return 0;
}
